### ISSUE TITLE
Can't resolve 'ws' in "puppeteer-core/lib"

### ISSUE THREAD

**Issue by vsvanshi** [saltyshiomix/nextron#65]
Hi Everyone,
I am opening this issue here, after spending long time on the internet to resolve the related issue but no luck.

THE ISSUE:
I am running an electron app with an example template of material ui and javascript.
Everything is working fine, but when I install the "puppeteer-core" library (version 2.0.0) and import it like 

`import puppeteer from "puppeteer-core";`

I start getting the following error

![Screenshot 2020-01-06 at 3 27 58 PM](https://user-images.githubusercontent.com/8156467/71810779-2f4e1f80-3099-11ea-9286-d5dc1baa559a.png)

Ofcourse then I started looking at SO and Google for this error and the "resolution" that everyone is suggesting is to use the following setting for webpack.config.js file

`const config = {
    // ...
    externals: {
        puppeteer: 'require("puppeteer")',
        // ...
    },
};`

Since nextron is based on next js, the place that I could find to put this setting was in 'next.config.js'
So I put the following code in it

`
...
config.externals = {
			puppeteer :'require("puppeteer")'
		}
...
		return config;`

But when I do this, A very strange error starts coming up.

![Screenshot 2020-01-06 at 3 32 53 PM](https://user-images.githubusercontent.com/8156467/71811078-d763e880-3099-11ea-97e5-b8c64ac549ae.png)

I am really not sure, what is going on here.

It would be really helpful if someone can just tell me what is going on, and how to resolve this.
Thanks in advance

**Comment by lacymorrow** [comment]
Are you still looking for help with this @vsvanshi?

**Comment by vsvanshi** [comment]
Yes, please @lacymorrow. I would definitely like to know what could have been done to prevent this error.

However for any user looking for a temporary solution, I had to leave nextron and proceed with "electron-webpack".

**Comment by lacymorrow** [comment]
Can you explain what you are trying to achieve with puppeteer inside
electron?

**Comment by vsvanshi** [comment]
yes Sure @lacymorrow 
i am making a desktop application which enables a user to open a specific website using puppeteer and and run a testing script on that website and see it all in action

**Comment by lacymorrow** [comment]
Thanks @vsvanshi, I may have some time in the next couple of weeks to look into this

**Comment by vsvanshi** [comment]
@lacymorrow did you get a chance to look at this?

**Comment by philipmckenna** [comment]
@vsvanshi I ran into a similar issue, however I was able to resolve it by adding to the externals rather than overriding them.

`... config.externals = [...config.externals, {whatever you wish to add}] ...`

**Comment by Psycokwet** [comment]
Hello there ! I tried to look into the use of puppeteer within an electron app, did you try https://www.npmjs.com/package/puppeteer-in-electron this package ? 

I did not yet made it work, but I think that's a better road so far. I might try to look into it further and work on an example with it when I can. 

Let me know if there is any update about it !
